packages/generate/src/syntax/toEdgeQL.ts

```diff
@@ -1395,6 +1398,9 @@ function shapeToEdgeQL(
   if (lines.length === 0 && injectImplicitId) {
     addLine("id");
   }
+  if (hasPolyEl && !seen.has("__typename")) {
+    addLine("__typename := .__type__.name");
+  }
```

Comment:
The new functionality implies a `__typename` field when needed. If the user specifies this and it doesn't match what is assumed then the types will be wrong. We could check the `__typename` value here if it is already present (from a user) and confirm it is `.__type__.name` or throw.
I couldn't figure out a clean way to do this because the actual generated EdgeQL explicitly references the scope, which I couldn't figure out how to access.
```edgeql
single __typename := __scope_1_defaultMovie.__type__.name,
```